hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/resolver/MembershipNamenodeResolver.java

```diff
@@ -478,4 +478,37 @@ private List<MembershipState> getRecentRegistrationForQuery(
   public void setRouterId(String router) {
     this.routerId = router;
   }
+
+  /**
+   * Shuffle cache, to ensure that the current nn will not be accessed first next time.
+   *
+   * @param nsId name service id
+   * @param namenode namenode contexts
+   */
+  @Override
+  public synchronized void shuffleCache(String nsId, FederationNamenodeContext namenode) {
+    cacheNS.compute(Pair.of(nsId, false), (ns, namenodeContexts) -> {
+      if (namenodeContexts != null && namenodeContexts.size() > 1) {
```

Comment:
You can do early exit with the reverse:
```
if (namenodeContexts == null || namenodeContexts.size() <= 1) {
  return namenodeContexts;
}
XXXX
```

Reply:
Thank you for your review, but I mistakenly submitted this pr to the community.
All your revision suggestions this time, I have submitted to this correct pr #5990 . you can review on this pr